packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -152,15 +153,15 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
         this.contextKeyService.onDidChange(e => {
             for (const [, view] of this.views.values()) {
                 const clauseContext = this.viewClauseContexts.get(view.id);
-                if (clauseContext && e.affects(clauseContext)) {
+                if (this.shouldUpdateViewVisibility(clauseContext, e)) {
```

Comment:
The logic of the new `shouldUpdateViewVisibility()` method is pretty different from the old logic.
The old logic passed if
- `clauseContext` existed 
- and `monaco` said that the change event affected one of the keys involved in the `clauseContext`.
 
The new logic passes if
- `clauseContext` does not exist (! big change)
-  or if it's `'true'` 
- or if `monaco` says so. 
 
It seems to me that if the `clauseContext` is `undefined`, we don't need to update (no key change could possibly affect it); if it's `'true'` we don't need to update the visibility - again no change could change `true` to anything else, so we only need to update `if (clauseContext instanceof Set && e.affects(clauseContext))`.

If that's correct, then since we're never going to update views that have their `when` set to `true` or `false`, we can just change l. 302 to `if (view.when && view.when !== 'false' && view.when !== 'true')`, keep the change at line 211 (`return view.when === undefined || view.when === 'true' || this.contextKeyService.match(view.when);`), and get rid of all the other changes. Does that cover all the bases?